src/stores/video.ts

```diff
@@ -222,45 +304,82 @@ export const useVideoStore = defineStore('video', () => {
         const canvas = document.createElement('canvas')
         const context = canvas.getContext('2d')
         if (!context) {
           URL.revokeObjectURL(videoObjectUrl)
-          reject('2D context not available.')
-          return
+          return reject('2D context not available.')
         }
 
         const [width, height] = [660, 370]
         canvas.width = width
         canvas.height = height
 
         video.currentTime = 0
-        seekResolve = () => {
-          context.drawImage(video, 0, 0, width, height)
-          const blobCallback = (blob: Blob | null): void => {
-            if (!blob) {
-              reject('Failed to create blob')
-              return
+        const tryExtractFrame = (time: number): Promise<Blob> => {
+          return new Promise((resolveFrame, rejectFrame) => {
+            const onSeeked = (): void => {
+              try {
+                context.drawImage(video, 0, 0, width, height)
+                canvas.toBlob(
+                  (blob: Blob | null) => {
+                    video.removeEventListener('seeked', onSeeked)
+                    if (blob) {
+                      resolveFrame(blob)
+                    } else {
+                      rejectFrame('Failed to create blob')
+                    }
+                    URL.revokeObjectURL(videoObjectUrl)
+                  },
+                  'image/jpeg',
+                  0.6
+                )
+              } catch (err) {
+                video.removeEventListener('seeked', onSeeked)
+                URL.revokeObjectURL(videoObjectUrl)
+                rejectFrame(err)
+              }
             }
-            resolve(blob)
-            URL.revokeObjectURL(videoObjectUrl)
-          }
-          canvas.toBlob(blobCallback, 'image/jpeg', 0.6)
+            video.addEventListener('seeked', onSeeked)
+            video.currentTime = time
+          })
         }
+
+        // First try extracting at time 0
+        tryExtractFrame(0)
+          .then(resolve)
+          .catch((err) => {
+            console.warn('Failed to extract thumbnail at t=0, trying at t=1:', err)
+            // If that fails, try at 1 second.
+            tryExtractFrame(1)
+              .then(resolve)
+              .catch((err2) => {
+                console.warn('Failed to extract thumbnail at t=1, using fallback thumbnail:', err2)
+                // Create a fallback thumbnail (a blank black image).
+                context.fillStyle = '#000'
+                context.fillRect(0, 0, width, height)
+                canvas.toBlob(
+                  (fallbackBlob: Blob | null) => {
+                    if (fallbackBlob) {
+                      resolve(fallbackBlob)
+                    } else {
+                      resolve(new Blob())
+                    }
+                    URL.revokeObjectURL(videoObjectUrl)
+                  },
+                  'image/jpeg',
+                  0.6
+                )
```

Comment:
Same here.

Try to avoid those infinite indentation hell at all cost. It makes reading the code an homeric task.